io/router/key.go

```diff
@@ -38,6 +40,24 @@ func (q *keyQueue) InputState() TextInputState {
 	return q.state
 }
 
+// InputState returns the last input mode state as
+// determined in Frame. It returns false if no change
+// happens.
+func (q *keyQueue) InputMode() (key.InputMode, bool) {
+	if q.focus == nil {
+		return key.KeyboardDefault, false
```

Comment:
I would expect `q.mode, false` here. See also my comment below for (*Router).TextInputMode.